test/test_dense/test_construction.cc

```diff
@@ -336,3 +337,139 @@ TEST_F(TestFillAxis, long_axis_2x4) {
+    ) {
+        // Allocate workspace
+        int64_t size = n_rows * n_cols;
+        std::vector<T> A(size, 0.0);
+        std::vector<T> B(size, 0.0);
+
+        // Construct the sketching operator
+        RandBLAS::DenseDist D = {
+            .n_rows = n_rows,
+            .n_cols = n_cols,
+            .family = dn
+        };
+
+        auto state = RandBLAS::RNGState(key);
+        auto next_state = RandBLAS::fill_dense(D, A.data(), state);
+        RandBLAS::fill_dense(D, B.data(), next_state);
+
+        ASSERT_TRUE(!(A == B));
+    }
+
+    template <typename T>
+    static void test_identity(
+        uint32_t key,
+        int64_t n_rows,
+        int64_t n_cols,
+        RandBLAS::DenseDistName dn
+    ) {
+        // Allocate workspace
+        int64_t size = n_rows * n_cols;
+        std::vector<T> A(size, 0.0);
+        std::vector<T> B(size, 0.0);
+
+        double total = 0.0;
+
+        // Construct the sketching operator
+        RandBLAS::DenseDist D1 = {
+            .n_rows = n_rows,
+            .n_cols = n_cols / 2,
+            .family = dn
+        };
+
+        RandBLAS::DenseDist D3 = {
+            .n_rows = n_rows,
+            .n_cols = n_cols - n_cols / 2,
+            .family = dn
+        };
+
+        RandBLAS::DenseDist D2 = {
+            .n_rows = n_rows,
+            .n_cols = n_cols,
+            .family = dn
+        };
+
+        auto state = RandBLAS::RNGState(key);
+        auto state1 = RandBLAS::RNGState(key);
+
+        // Concatenates two matrices generated from state and next_state
+        auto next_state = RandBLAS::fill_dense(D1, A.data(), state);
+        RandBLAS::fill_dense(D3, A.data() + (int64_t) (D1.n_rows * D1.n_cols), next_state);
+
+        RandBLAS::fill_dense(D2, B.data(), state1);
+
+        for (int i = 0; i < size; i++) {
+            total += abs(A[i] - B[i]);
+        }
+        
+        ASSERT_TRUE(total == 0.0);
+    }
+    
+    /*template<typename T>
+    static void test_identity_trunc(
+        uint32_t key,
+        int64_t n_rows,
+        int64_t n_cols
+    ) {
+        // Allocate workspace
+        int64_t size = n_rows * n_cols;
+        std::vector<T> A(size, 0.0);
+        std::vector<T> B(size, 0.0);
+        double total = 0;
+
+        int64_t n_srows1 = n_rows / 2;
+        int64_t n_srows2 = n_rows - n_srows1;
+        int64_t ptr = n_srows1 * n_cols;
+
+        auto state = RandBLAS::RNGState(key);
+
+        auto next_state = RandBLAS::fill_dense_submat_impl<T, r123::Philox4x32,r123ext::boxmul>(n_cols, A.data(), n_srows1, n_cols, 0, state);
+        RandBLAS::fill_dense_submat_impl<T, r123::Philox4x32,r123ext::boxmul>(n_cols, A.data() + ptr, n_srows2, n_cols, 0, next_state);
+
+        RandBLAS_Testing::Util::fill_dense_rmat_trunc<T, r123::Philox4x32,r123ext::boxmul>(B.data(), n_rows, n_cols, state);
+
+        for (int i = 0; i < size; i++) {
+            total += abs(A[i] - B[i]);
+        }
+
+        ASSERT_TRUE(total == 0);
+
+    }*/
+};
```

Comment:
@kaiwenhe7 is this test commented out deliberately?

Reply:
@rileyjmurray Yes. I commented it out because the original test_identity works now using fill_dense.

Reply:
Deleting it would be the right move here

Reply:
Cool done!